kythe/cxx/indexer/cxx/KytheGraphObserver.cc

```diff
@@ -870,6 +872,20 @@ void KytheGraphObserver::recordFunctionNode(
   }
 }
 
+void KytheGraphObserver::assignUsr(const NodeId& node, llvm::StringRef usr,
+                                   int byte_size) {
+  usr = usr.take_front(byte_size);
```

Comment:
What happens if byte_size is larger than the usr?

Reply:
As many bytes as are available are returned.

Reply:
minor: IMO, the `byte_size` argument is redundant here; this is kind of the purpose of StringRef/string_view classes and I'd rather have the caller make this adjustment if necessary.